### PR TITLE
Next gen

### PR DESCRIPTION
**Pull request by joliveros**
- Fix http server onListen callback, use node Promises instead of bluebird
- Delete cached model definition Everytime the Ticket model is defined
- Update npm scripts
- Update npm dependencies
- Updates routes/token.js to use new twilio-node api
- Update test/connectionHelper.js to throw if conn string is not defined, Attempts to connect only once.
- Fix POST /tickets/new test
- Update twilio browser client sdk version, remove calls to deprecated methods
- Update html5shiv script
- Add MONGO_URL to travis.yml
- Update travis.yml to enable compilation of native extensions
- Update/fix redirect on new ticket submit. Add notification on success/failure
- Add SUPPORT_NUMBER env var to be used in POST /call/connect
- Update .gitignore to ignore .env.local files
- Update readme header and clarify local development steps
- Update package.json dependencies
- Fix broken tests
- Fix jade compiler warnings
- Fix travis.yml syntax error

### CODE REVIEW

**Comment by joliveros** [comment]
@acamino this PR is ready for your review.

Cheers 🎉

File: .gitignore [hunk 1]
```diff
@@ -31,3 +31,6 @@ node_modules
 
 # Optional REPL history
 .node_repl_history
+
+# local env variables
+.env.local
```
Comment (juancarlospaco) [inline .gitignore]: We should be using `.env` as the filename for the local env variables according to the instructions.

File: README.md [hunk 2]
```diff
@@ -1,101 +1,96 @@
...
+    git clone git@github.com:TwilioDevEd/browser-calls-node.git
 
-   cd browser-calls-node
-   ```
+    cd browser-calls-node
+    ```
 
 1. Install dependencies:
-
     ```
```
Comment (juancarlospaco) [inline README.md]: ditto.

File: README.md [hunk 3]
```diff
@@ -1,101 +1,96 @@
...
-
-To run the app locally, clone this repository and `cd` into its directory:
+First you need to install
+  - [Node.js](http://nodejs.org/) which should also install [npm](https://www.npmjs.com/).
+  - [MongoDB](https://www.mongodb.org/)
 
 1. First clone this repository and `cd` into its directory:
-   ```
-   git clone git@github.com:TwilioDevEd/browser-calls-node.git
+    ```
```
Comment (juancarlospaco) [inline README.md]: use `bash` as language for markup highlight.

File: public/js/browser-calls.js [hunk 4]
```diff
@@ -108,3 +99,51 @@ function callSupport() {
...
   Twilio.Device.disconnectAll();
 }
+
+function initNewTicketForm() {
+  var formEl = $(".new-ticket");
+  var buttonEl = formEl.find(".btn.btn-primary");
+
+  // button handler
+  formEl.find("[type='button']").click(function(e) {
+    // e.preventDefault()
```
Comment (juancarlospaco) [inline public/js/browser-calls.js]: Remove commented out code.

File: routes/token.js [hunk 5]
```diff
@@ -6,7 +6,7 @@ var twilio = require('twilio');
 router.post('/generate', function (req, res) {
   var page = req.body.page;
 
-  var capability = new twilio.Capability(
+  var capability = twilio.jwt.Capability(
```
Comment (juancarlospaco) [inline routes/token.js]: is it missing a `new` here ?

File: package.json [hunk 6]
```diff
@@ -3,8 +3,8 @@
   "version": "1.0.0",
   "private": true,
   "scripts": {
-    "start": "node ./bin/www",
-    "test": "export NODE_ENV=test && mocha test"
+    "start": "DEBUG=test-app* node ./bin/www",
```
Comment (joliveros) [inline package.json]: The tutorial uses https://www.npmjs.com/package/debug to log the server listening events. So if you run the server without the debug logging enabled you won't see that the server is running.
https://github.com/TwilioDevEd/browser-calls-node/blob/next-gen/bin/www#L82

File: public/js/browser-calls.js (hunk 4 above)
Comment (joliveros) [inline public/js/browser-calls.js]: nice catch!

File: public/js/browser-calls.js [hunk 7]
```diff
@@ -108,3 +99,51 @@ function callSupport() {
...
+    $.ajax({
+        url: '/tickets/new',
+        type: 'post',
+        data: formEl.serialize()
+    })
+    .fail(function(res) {
+      showNotification("Support ticket request failed. " + res.responseText, "danger")
+    })
+    .done(function(){
```
Comment (juancarlospaco) [inline public/js/browser-calls.js]: The `.done( )` should be first then the `.fail( )`.

**Comment by joliveros** [comment]
@juancarlospaco, thanks for the review. Please find the commits addressing the issues you found here. 😉

**Comment by acamino** [comment]
LGTM!

**Pull request merged**